Docs (#11)
Observatory-Sciences/aravis-detector: README.md

  - [ZeroMQ](https://zeromq.org/) asynchronous message library.
  - [Log4CXX](https://logging.apache.org/log4cxx/latest_stable/)  message logs.
  - [HDF5](https://www.hdfgroup.org/HDF5) used in the file saver plugin.
- Aravis:
  - [Meson](https://mesonbuild.com/): build system
  - [GLib - 2.0](https://docs.gtk.org/glib/)
  - [GObject - 2.0](https://docs.gtk.org/gobject/)

## Install

To build all sub-dependencies you can run the following in your shell:

Odin Data:

```shell
sudo apt -y update
sudo apt install cmake python3.10-venv
sudo apt install libboost-program-options-dev libboost-filesystem-dev \
 libboost-date-time-dev libboost-dev libboost-system-dev \
 libboost-test-dev libboost-thread-dev libboost-regex-dev \
 libzmq3-dev libpcap-dev liblog4cxx-dev libblosc-dev \
 libhdf5-dev librdkafka-dev
```

Aravis:

```shell
sudo apt install ninja-build build-essential meson libxml2-dev libglib2.0-dev \
 libusb-1.0-0-dev gobject-introspection libgtk-3-dev \
 gtk-doc-tools xsltproc libgstreamer1.0-dev \
 libgstreamer-plugins-base1.0-dev \
 libgstreamer-plugins-good1.0-dev\
 libgirepository1.0-dev gettext
```

### Building Dependencies

> (Optional) It's useful to save the plugin an it's dependencies in same folder and create a virtual environment for the python server. Create a shortcut to this folder under the name "cwd" as the rest of the guide utilizes it. Additionally you can create a temp folder for all the saved files. To do all this navigate to your desired location for the software and run the following:

Reviewer: ```suggestion
(Optional) It's useful to save the plugin an its dependencies in same folder and create a virtual environment for the python server. Create a shortcut to this folder under the name "ODIN" as the rest of the guide utilizes it. Additionally you can create a temp folder for all the saved files. To do all this navigate to your desired location for the software and run the following:
```

The name "cwd" is not a great choice. It generally means "Current Working Dir" and is dynamic. Instead chose a more descriptive name. In the above change suggestion, I suggest "ODIN" but something else may be better?

```shell
mkdir odin_camera_driver && cd odin_camera_driver

> cwd=$(pwd)

Reviewer: ```suggestion
```

This is not necessary - or if you want to store the `pwd` for later then store it in a named variable instead.

https://observatory-sciences.github.io/aravis-detector/

---

- [Aravis detector](#aravis-detector)
  - [Dependencies](#dependencies)
  - [Install](#install)
    - [Building Dependencies](#building-dependencies)
      - [Build Odin-data](#build-odin-data)
      - [Install Odin Control](#install-odin-control)
      - [Build Aravis](#build-aravis)
      - [Build aravis-detector](#build-aravis-detector)
  - [Docs](#docs)
  
---

## Dependencies

The plugin has the following direct dependencies:

- [Aravis v0.8.31](https://github.com/AravisProject/aravis): used to interface with genicam cameras.

> - [Cmake v2.9](https://cmake.org/): used to generate the make files.

Reviewer: ```suggestion
- [Cmake >= v2.9](https://cmake.org/): used to generate the make files.
```

Modern cmake is > v3.0 and cmake is backwards compatible. Generally users of cmake should always choose the very latest version of cmake because it will work with the latest compiler versions. So, I don't think its good practice to suggest an exact version of cmake and certainly not as old as 2.9. Rather specify a minimum version...

Is this Odin that uses such an old version of cmake or where did v2.9 come from?
